### PR TITLE
plat-sam: Set QSPI memories as non secure

### PR DESCRIPTION
**Pull request by clementleger**
plat-sam: set QSPI memories as non secure

When left unconfigured, the QSPI memories are in the secure world.
However, the controller is not secure and Linux expect to use these
memories with them and it will fail because the memories are not
accessible. Configure them as non secure to let linux handle the QSPI
controller.

Signed-off-by: Clément Léger <clement.leger@bootlin.com>

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
For the commit log: "memories are in the secure world": maybe rephrase as "memories are assigned to secure world". (or "only accessible from").
Also: s/expect/expects/, remove "with them", s/linux/Linux/ (preferably non-secure world, normal world or REE).

**Comment by clementleger** [comment]
New version:
- Reword and fix comment

File: core/arch/arm/plat-sam/main.c [hunk 1]
```diff
@@ -176,8 +176,24 @@ static void matrix_configure_slave_h64mx(void)
...
 	/* 11:  Internal SRAM 128K (Cache L2): Default */
-	/* 12:  QSPI0: Default */
-	/* 13:  QSPI1: Default */
+
+	/* 12:  QSPI0: Normal world */
+	/* 13:  QSPI1: Normal world */
+	srtop_setting = MATRIX_SRTOP(0, MATRIX_SRTOP_VALUE_128M);
+	sasplit_setting = MATRIX_SASPLIT(0, MATRIX_SASPLIT_VALUE_128M);
+	ssr_setting = (MATRIX_LANSECH_NS(0) | MATRIX_RDNSECH_NS(0) |
+		       MATRIX_WRNSECH_NS(0));
```
Comment (etienne-lms) [inline core/arch/arm/plat-sam/main.c:185]: could remove outer parentheses.

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by clementleger** [comment]
Any other comments ?

File: core/arch/arm/plat-sam/main.c [hunk 2]
```diff
@@ -176,8 +176,24 @@ static void matrix_configure_slave_h64mx(void)
...
+	srtop_setting = MATRIX_SRTOP(0, MATRIX_SRTOP_VALUE_128M);
+	sasplit_setting = MATRIX_SASPLIT(0, MATRIX_SASPLIT_VALUE_128M);
+	ssr_setting = MATRIX_LANSECH_NS(0) | MATRIX_RDNSECH_NS(0) |
+		      MATRIX_WRNSECH_NS(0);
+
+	matrix_configure_slave_security(matrix64_base(),
+					H64MX_SLAVE_QSPI0,
+					srtop_setting,
+					sasplit_setting,
+					ssr_setting);
```
Comment (jforissier) [inline core/arch/arm/plat-sam/main.c:191]: Preferred indentation style:
```C
	matrix_configure_slave_security(matrix64_base(),
					H64MX_SLAVE_QSPI0, srtop_setting,
					sasplit_setting, ssr_setting);         
```
(Linux kernel coding style: "Descendants are always substantially shorter than the parent and are placed substantially to the right. A very commonly used style is to align descendants to a function open parenthesis.")

**Review by jforissier — COMMENTED** [review]
On nit below, and in the commit description: s/uis/is/.
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by jforissier** [comment]
@clementleger you pushed an identical commit is seems :wink:

**Comment by clementleger** [comment]
> @clementleger you pushed an identical commit is seems wink

Old one is this one: https://github.com/OP-TEE/optee_os/commit/55c3ddba565f6a29c1e1d85e667e483484b04c92
New one is this one: https://github.com/OP-TEE/optee_os/commit/1a548dc0bcdd061fd841c7358f7fb648279e8453

There seems to be a difference to me but I might be wrong !

**Comment by jforissier** [comment]
@clementleger OK I think the diff rendering of GitHub messes indentation a bit, anyway the code does look good now. But, you didn't fix the typo (s/uis/is) nor added my Ack :cry:

**Comment by clementleger** [comment]
> @clementleger OK I think the diff rendering of GitHub messes indentation a bit, anyway the code does look good now. But, you didn't fix the typo (s/uis/is) nor added my Ack cry

Ahem, sorry, start of year is tough... Here you go. And indeed, github messed up the indentation.

**Comment by jforissier** [comment]
@clementleger all good now, thanks!
